app/scripts/lib/sentry.js

```diff
@@ -0,0 +1,133 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+define([
+  'underscore',
+  'raven',
+  'lib/url'
+], function (_, Raven, Url) {
+
+  var ALLOWED_QUERY_PARAMETERS = [
+    'client_id',
+    'keys',
+    'redirect_uri',
+    'scope',
+    'style',
+    'verification_redirect',
+    'campaign',
+    'customizeSync',
+    'entrypoint',
+    'context',
+    'migration',
+    'service',
+    'setting',
+    'automatedBrowser'
+  ];
+
+  /**
+   * function that gets called before data gets sent to error metrics
+   *
+   * @param {Object} data
+   *  Error object data
+   * @returns {Object} data
+   *  Modified error object data
+   * @private
+   */
+  function beforeSend(data) {
+    if (data && data.request && data.request.url) {
+      data.request.url = cleanUpQueryParam(data.request.url);
+    }
+
+    return data;
+  }
+
+  /**
+   * Sanitize and clear extra query parameter information
+   *
+   * @param {String} url
+   * @returns {String} url
+   * @private
+   */
+  function cleanUpQueryParam(url) {
+    var startOfParams = url.indexOf('?');
+    var newUrl = url;
+    var params;
+
+    if (startOfParams >= 0) {
+      params = Url.searchParams(url.substring(startOfParams + 1));
+      newUrl = url.substring(0, startOfParams);
+    }
+
+    if (_.isObject(params)) {
```

Comment:
Can lines 64->73 be moved above the closing brace on line 62? IIUC, the evaluation made here can only be true if `startOfParams >= 0`.

Or, as an alternative, you could use a guard statement higher up:

```js
if (startOfParams === -1) {
  return url;
}

params = Url.searchParams(...
```